tests/units/units.cpp

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2015 University of Auckland
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.Some license of other
+*/
+
+#include "gtest/gtest.h"
+
+#include <libcellml>
+
+
+TEST(Units, serialise) {
+    const std::string e = "<units/>";
+    libcellml::Units u;
+    std::string a = u.serialise(libcellml::CELLML_FORMAT_XML);
+    EXPECT_EQ(e, a);
+}
+
+TEST(Units, valid_name) {
+    const std::string e =
+            "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+            "<model xmlns=\"http://www.cellml.org/cellml/1.2#\">"
+            "<units name=\"valid_name\" base_unit=\"yes\"/>"
+            "</model>";
+
+    libcellml::Model m;
+
+    libcellml::UnitsPtr u = std::make_shared<libcellml::Units>();;
+    u->setName("valid_name");
+
+    /* Base unit is false by default */
+    EXPECT_FALSE(u->isBaseUnit());
+    u->setBaseUnit();
+    EXPECT_TRUE(u->isBaseUnit());
+    u->setBaseUnit(false);
+    EXPECT_FALSE(u->isBaseUnit());
+    u->setBaseUnit(true);
+    EXPECT_TRUE(u->isBaseUnit());
```

Comment:
line 43 sets the base unit flag true, then we turn it off again and check that the flag is false and finally turn the flag on again in a different way so that the unit is output with the base unit flag set.